packages/odyssey-react-mui/src/FieldComponentProps.ts

```diff
@@ -55,7 +53,30 @@ export type FieldComponentProps = {
    * The name of the `input` element. Defaults to the `id` if not set.
    */
   name?: string;
-};
+} & (
+  | {
+      /**
+       * If `hasVisibleLabel` is `false` it will also hide any `hint` text and `HintLinkComponent` that is passed in
+       */
+      hasVisibleLabel?: false;
+      hint: never;
+      HintLinkComponent: never;
+    }
+  | {
+      /**
+       * defaults to `true` if `false` along with hiding the label it will also hide any `hint` text and `HintLinkComponent` that is passed in
+       */
+      hasVisibleLabel?: true;
+      /**
+       * The helper text content.
+       */
+      hint?: string;
+      /**
+       * A `Link` component to provide greater context that is rendered at the end of the `hint` text
+       */
+      HintLinkComponent?: ReactElement<typeof HintLink>;
+    }
+);
```

Comment:
Is it possible we could rewrite this type to use the function overload syntax? Or have 2 different `Field` components, one with a label and one that only takes `aria-labelledby` or something like that?

The way the code works today, it's very confusing.

Reply:
This would be a pretty significant rewrite. We'd have to refactor all the form field components(that allow a non-visible label) to handle this 'correctly'